### PR TITLE
Add an option to use Docker instead of Podman

### PR DESCRIPTION
**Pull request by maelk**
Creates the environment variable CONTAINER_RUNTIME . The variable can be set to "docker" or "podman" depending on the runtime you want to use. It defaults to Podman. The requested runtime will be installed automatically.

### CODE REVIEW

File: host_cleanup.sh [hunk 1]
```diff
@@ -6,13 +6,13 @@ source lib/common.sh
...
-    sudo podman ps --all | grep -w "$name$" && sudo podman rm $name -f
+    sudo ${CONTAINER_RUNTIME} ps | grep -w "$name$" && sudo ${CONTAINER_RUNTIME} kill $name
+    sudo ${CONTAINER_RUNTIME} ps --all | grep -w "$name$" && sudo ${CONTAINER_RUNTIME} rm $name -f
 done
 
 # Remove existing pod
-if  sudo podman pod exists ironic-pod ; then
-    sudo podman pod rm ironic-pod -f
+if  sudo ${CONTAINER_RUNTIME} pod exists ironic-pod ; then
+    sudo ${CONTAINER_RUNTIME} pod rm ironic-pod -f
```
Comment (russellb) [inline host_cleanup.sh]: Should this be run with podman only like pod creation?

File: 02_configure_host.sh [hunk 2]
```diff
@@ -61,23 +61,23 @@ else
       fi
       sudo ifdown provisioning || true
       sudo ifup provisioning
-  
+
```
Comment (russellb) [inline 02_configure_host.sh]: Nit: ideally unrelated formatting changes would be split out into another commit. You can leave it though

File: host_cleanup.sh (hunk 1 above)
Comment (evrardjp) [reply host_cleanup.sh]: Agreed with @russellb .

Comment (maelk) [reply host_cleanup.sh]: fixed, thank you

File: 02_configure_host.sh (hunk 2 above)
Comment (maelk) [reply 02_configure_host.sh]: ok, sorry

**Comment by russellb** [comment]
tested, still works fine in my env

**Comment by sb1975** [comment]
So where do I specify CONTAINER_RUNTIME ?
By default will it install docker now ?
Can we update the main document to clarify this please ?

**Comment by russellb** [comment]
> So where do I specify CONTAINER_RUNTIME ?

config_$USER.sh

> By default will it install docker now ?

No, still podman by default

> Can we update the main document to clarify this please ?

Patches welcome the README :-)
